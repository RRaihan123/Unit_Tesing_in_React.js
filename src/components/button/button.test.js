import React from 'react';
import {render,screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from './button'
it("button renders correctly1",()=>{
    const {getByTestId} = render(<Button label="click me please"></Button>);
    expect(getByTestId('button1')).toHaveTextContent(/^hey click me please$/);
})
it("button renders correctly2",()=>{
    render(<Button label="Hi"></Button>);
    expect(screen.getByTestId('button1')).toHaveTextContent(/^hey Hi$/);
})
it("button renders correctly3",()=>{
    const {getByTestId} = render(<Button label="i am the button"></Button>);
    expect(getByTestId('button1')).toHaveTextContent(/^hey i am the button$/);
})

it("matches snapshot 1",()=>{
    const tree = renderer.create(<Button label="button1"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("matches snapshot 2",()=>{
    const tree = renderer.create(<Button label="button2"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})




