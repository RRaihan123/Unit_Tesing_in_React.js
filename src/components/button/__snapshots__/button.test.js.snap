// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button1"
>
  hey 
  button1
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button1"
>
  hey 
  button2
</div>
`;
